opm/autodiff/AquiferFetkovich.hpp

```diff
@@ -0,0 +1,387 @@
+        std::vector<Eval> rhow_;
+        std::vector<Scalar> alphai_;
+        std::vector<int> cellToConnectionIdx_;
+
+        // Variables constants
+        const Aquifetp::AQUFETP_data aqufetp_data_;
+        const Aquancon::AquanconOutput connection_;
+
+        Scalar mu_w_; //water viscosity
+        Scalar Tc_;   // Time Constant
+        Scalar pa0_;    // initial aquifer pressure
+        Scalar aquifer_pressure_; // aquifer pressure
+
+        Eval W_flux_;
+
+        Scalar gravity_() const
+        { return ebos_simulator_.problem().gravity()[2]; }
+
+        inline void initQuantities(const Aquancon::AquanconOutput& connection)
+        {
+          // We reset the cumulative flux at the start of any simulation, so, W_flux = 0
+          W_flux_ = 0.;
+
+          // We next get our connections to the aquifer and initialize these quantities using the initialize_connections function
+          initializeConnections(connection);
+
+          calculateAquiferCondition();
+
+          pressure_previous_.resize(cell_idx_.size(), 0.);
+          pressure_current_.resize(cell_idx_.size(), 0.);
+          Qai_.resize(cell_idx_.size(), 0.0);
+        }
+
+        inline void updateCellPressure(std::vector<Eval>& pressure_water, const int idx, const IntensiveQuantities& intQuants)
+        {
+          const auto& fs = intQuants.fluidState();
+          pressure_water.at(idx) = fs.pressure(waterPhaseIdx);
+        }
+
+        inline void updateCellPressure(std::vector<Scalar>& pressure_water, const int idx, const IntensiveQuantities& intQuants)
+        {
+          const auto& fs = intQuants.fluidState();
+          pressure_water.at(idx) = fs.pressure(waterPhaseIdx).value();
+        }
+
+        inline void updateCellDensity(const int idx, const IntensiveQuantities& intQuants)
+        {
+          const auto& fs = intQuants.fluidState();
+          rhow_.at(idx) = fs.density(waterPhaseIdx);
+        }
+
+        inline Scalar dpai(int idx)
+        {
+          Scalar dp = aquifer_pressure_ + rhow_.at(idx).value()*gravity_()*(cell_depth_.at(idx) - aqufetp_data_.d0) - pressure_current_.at(idx).value() ;
+          return dp;
+        }
+        // This function implements Eq 5.12 of the EclipseTechnicalDescription
+        inline Scalar aquiferPressure()
+        {
+          Scalar Flux = W_flux_.value();
+          Scalar pa_ = pa0_ - Flux / ( aqufetp_data_.C_t * aqufetp_data_.V0 );
+          return pa_;
+        }
+        // This function implements Eq 5.14 of the EclipseTechnicalDescription
+        inline void calculateInflowRate(int idx, const Simulator& simulator)
+        {
+          Tc_ = ( aqufetp_data_.C_t * aqufetp_data_.V0 ) / aqufetp_data_.J ;
+          Scalar td_Tc_ = simulator.timeStepSize() / Tc_ ;
+          Scalar exp_ = (1 - exp(-td_Tc_)) / td_Tc_;
+          Qai_.at(idx) = alphai_.at(idx) * aqufetp_data_.J * dpai(idx) * exp_;
+        }
+
+        template<class faceCellType, class ugridType>

```

Comment:
This function looks like the same with the same function from AquiferCarterTracy.hpp, and it does not use any member or functions from the class. Maybe you can put this type of function to header file like `AquiferHelpers.hpp`. 

You can refer to `WellHelpers.hpp`.